lib/cloud/imds/azure/imds_test.go

```diff
@@ -172,6 +172,59 @@ func TestParseMetadataClientError(t *testing.T) {
 	}
 }
 
+func TestGetInstanceInfo(t *testing.T) {
+	for _, tc := range []struct {
+		name                 string
+		stausCode            int
+		body                 []byte
+		expectedInstanceInfo *InstanceInfo
+		errAssertion         require.ErrorAssertionFunc
+	}{
+		{
+			name:      "with resource ID",
+			stausCode: http.StatusOK,
+			body:      []byte(`{"resourceId":"test-id"}`),
+			expectedInstanceInfo: &InstanceInfo{
+				ResourceID: "test-id",
+			},
+			errAssertion: require.NoError,
+		},
+		{
+			name:      "all fields",
+			stausCode: http.StatusOK,
+			body: []byte(`{"resourceId":"test-id", "location":"eastus", "resourceGroupName":"TestGroup", ` +
+				`"subscriptionId": "5187AF11-3581-4AB6-A654-59405CD40C44", "vmId":"ED7DAC09-6E73-447F-BD18-AF4D1196C1E4"}`),
+			expectedInstanceInfo: &InstanceInfo{
+				ResourceID:        "test-id",
+				Location:          "eastus",
+				ResourceGroupName: "TestGroup",
+				SubscriptionID:    "5187AF11-3581-4AB6-A654-59405CD40C44",
+				VMID:              "ED7DAC09-6E73-447F-BD18-AF4D1196C1E4",
+			},
+			errAssertion: require.NoError,
+		},
+		{
+			name:         "request error",
+			stausCode:    http.StatusNotFound,
+			errAssertion: require.Error,
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+				w.WriteHeader(http.StatusOK)
```

Comment:
Should this be:

```suggestion
				w.WriteHeader(tc.statusCode)
```

If yes, please double-check that the tests are working as intended.

Reply:
Thanks!
It failed anyway because there was no body.
I'll add a second test to assert that an empty body returns an error, as well as a non-OK status code.